### PR TITLE
added segment integration

### PR DESCRIPTION
**Pull request by chitalian**

### CODE REVIEW

File: valhalla/jawn/src/lib/handlers/SegmentLogHandler.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  public async handleResults(): PromiseGenericResult<string> {
+    for (const segmentEvent of this.segmentEvents) {
+      try {
+        await sendSegmentEvent(segmentEvent);
+      } catch (error) {
+        console.error(error);
+      }
+    }
+
+    return ok("Successfully handled lytix logs");
```
Comment (colegottdank) [inline valhalla/jawn/src/lib/handlers/SegmentLogHandler.ts:110]: segment not lytix

File: valhalla/jawn/src/lib/handlers/SegmentLogHandler.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  private mapSegmentEvent(
+    context: HandlerContext,
+    writeKey: string
+  ): SegmentEvent {
+    const request = context.message.log.request;
+    const response = context.message.log.response;
+    const usage = context.usage;
+
+    return {
+      // writeKey: context.message.heliconeMeta.segmentWriteKey,
```
Comment (colegottdank) [inline valhalla/jawn/src/lib/handlers/SegmentLogHandler.ts:122]: needed?

File: valhalla/jawn/src/lib/handlers/SegmentLogHandler.ts (hunk 1 above)
Comment (chitalian) [inline valhalla/jawn/src/lib/handlers/SegmentLogHandler.ts:110]:
```suggestion
    return ok("Successfully handled segment logs");
```

File: valhalla/jawn/src/lib/handlers/SegmentLogHandler.ts (hunk 2 above)
Comment (chitalian) [inline valhalla/jawn/src/lib/handlers/SegmentLogHandler.ts:122]:
```suggestion
```

nope

**Pull request merged**